### ISSUE TITLE
Footer being detected as header???

### ISSUE THREAD

**Issue by HysMX** [Secreto31126/whatsapp-api-js#233]
**Describe the bug**
When trying to construct a footer inside an interactive object it detects the footer as part of the header for some reason

**To Reproduce**
Just place " new Footer("text") inside the interactive object after the body

Error:
![image](https://github.com/Secreto31126/whatsapp-api-js/assets/50180189/2fc1caa2-9585-4c05-bf75-9f84a0d57f92)

I think its because its expecting the header to be defined after the body  but if i don't want to use a header, i think you should be able to construct the footer after the body with no issue.

This same problem happens if you try to define the header before the body which i think its kinda intuitive to do since you're constructing it from top to bottom

**Comment by Secreto31126** [comment]
The order of the elements was picked based on the likelihood to be used, so it's less likely to need using `undefined` in cases like yours:

 - Body is required in 5 out of 6 (from the top of my head) scenarios, the only exception is Single Product where it's forbidden.

 - Header is optional in most cases, except in two cases, one where it's required and one where it's forbidden.

 - Footer is always optional.

On the other hand, the main reason why it doesn't receive a spread of elements, such as Template's components argument, is documentation. Having the arguments splitted in 3 different names makes it much easier to refer to them, specially since Interactive has so many edge cases in the throws.

Last, and probably least, I don't know how to type an unsorted tuple, except for writing the 9 combinations with an union 😂.

So, TL;DR, it's a topic I had been thinking about for a while, and I still believe the current implementation is the best I can provide. A little unintuitive for the order and the need to use `undefined` in edge cases, but simple in the simple scenarios.

What do you think? Should I consider using an "unsorted tuple" at the expense of the documentation, change the order of the arguments to be more intuitive (would require a breaking update), or leave it as it is? I would love to know your opinion!

**Comment by HysMX** [comment]
i would leave it as is but maybe specify in the comments you can pass "undefined"  i can see at least in one of the examples for a list you pass undefined, i'm braindead and couldn't think to try to use "undefined" 

Here
<img width="1077" alt="Captura de pantalla 2023-09-28 a la(s) 0 03 10" src="https://github.com/Secreto31126/whatsapp-api-js/assets/50180189/f5227ef5-dcd6-4bb4-a455-7b2d5fa9febc">

**Comment by Secreto31126** [comment]
That's something we can work with. Adding a "it may be undefined if not needed" to each argument might make it much clearer to use.

**Comment by Secreto31126** [comment]
![Screenshot_20230928_073930_Chrome.jpg](https://github.com/Secreto31126/whatsapp-api-js/assets/46955459/9aefcb4a-a239-493c-b8f7-687b3d45c85d)
